src/data-operations/data-container.ts

```diff
@@ -50,4 +50,56 @@ export class DataContainer {
         this.transformedData = DataUtil.process(this.data, this.state);
         return this;
     }
+    protected accessData (dataAccess: DataAccess) {
+        var res;
+        switch(dataAccess) {
+            case DataAccess.OriginalData:
+               res = this.data;
+               break;
+            case DataAccess.TransformedData:
+                res = this.transformedData;
+                break;
+        }
+        return res;
+    }
+    // CRUD operations
+    // access data records
+    getRecordInfoByKeyValue (fieldName: string, value: any, dataAccess: DataAccess = DataAccess.OriginalData): RecordInfo {
+        var data = this.accessData(dataAccess),
+            len = data.length, i,
+            res:RecordInfo = {index: -1, record: undefined};
+        for (i = 0; i < len; i++) {
+            if (data[i][fieldName] === value) {
+                res.index = i;
+                res.record = data[i];
+                break;
+            }
+        }
+        return res;
+    }
+    addRecord (record: Object, at?: number, dataAccess: DataAccess = DataAccess.OriginalData): void {
+        var data = this.accessData(dataAccess);
```

Comment:
Add and delete in the container should always be on the original data, shouldn't it?